setup.py

```diff
@@ -519,21 +540,229 @@ def add_dir_name(dir_name, filenames):
 
 inference_extension = create_extension(**inference_extension_info)
 
+tick_modules = [array_extension, base_extension,
+                test_extension, random_extension, simulation_extension,
+                model_core, prox_core, solver_core,
+                inference_extension]
+
 
 class CustomBuild(build):
+    swig_min_ver = (3, 0, 7)
+
+    @staticmethod
+    def extract_swig_version(swig_ver_str):
+        m = re.search('SWIG Version (\d+).(\d+).(\d+)', swig_ver_str)
+
+        if not m:
+            txt = 'Could not extract SWIG version from string: {0}'
+
+            warnings.warn(txt.format(swig_ver_str))
+
+            return 0, 0, 0
+
+        return tuple(int(x) for x in m.groups()[0:3])
+
     def run(self):
+        swig_ver = self.extract_swig_version(
+            str(subprocess.check_output(['swig', '-version'])))
+
+        if swig_ver < self.swig_min_ver:
+            txt = 'SWIG version {0}.{1}.{2} ' \
+                  'lower than the required version >= {3}.{4}.{5}. ' \
+                  'This will likely cause build errors!'
+
+            warnings.warn(txt.format(*(swig_ver + self.swig_min_ver)))
+
         self.run_command('build_ext')
         build.run(self)
 
 
 class CustomInstall(install):
     def run(self):
         self.run_command('build_ext')
-        self.do_egg_install()
+        install.run(self)
+
+
+class TickCommand(Command):
```

Comment:
For a quicker understanding, we might document it as an abstract class for tick commands